inc/telemetry/namespace.php

```diff
@@ -507,21 +511,93 @@ function handle_telemetry_endpoint( WP_REST_Request $request ) {
  */
 function track_new_or_updated_content( $post_id, $post, $update ) {
 	// Bail on auto-save.
-	if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
+	if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE || 'auto-draft' === $post->post_status ) {
 		return;
 	}
```

Comment:
I have had to include the `auto-draft` status here in the return as it seems to apply when a new post ( the exact post_type ) is created but not other post types. Not returning then fires off the tracking event. Essentially firing two tracking events when creating a new post.

Reply:
I would do this in a separate if statement. Read through the logic here, I don't think it will do what you in every scenario. What if `DOING_AUTOSAVE` is defined but false. You should never mix `&&` operators directly with `||`.